### ISSUE TITLE
Bug/feature request: processFiles(), this function should skip the already processed files

### ISSUE THREAD

**Issue by kepi0809** [pqina/filepond#183]
Hey,

as I mentioned above, it would be cool to have a function that processes all files and skips the ones that have already been processed. I'm not sure whether it's a bug or a feature request (bit of both?)

And if I'm already here then I'd ask how I could do that instead of 'Undo' button there would be remove button? It would be cool if it'd be possible. For some reasons the @processfileundo nor the @processfileabort wasn't fired when I clicked 'undo' it only made a delete request to the url/edit.

Thanks in advance! Good work btw. 👍 

Best wishes,
Kepi

**Comment by rikschennink** [comment]
Good point, this is probably a bug, will look into this as soon as possible.

**Comment by kepi0809** [comment]
processfileundo doesn't work on vue for me, tried both ways as below. What I would like to have that when the user clicks on undo then it would delete the file (so I'd actually call onRemoveFile for fileundo)

```
  <FilePond
    ref="pond"
    :label-idle="labelIdle"
    :accepted-file-types="acceptedFileTypes"
    :allow-multiple="allowMultiple"
    :instantUpload="instantUpload"
    :allowReplace="!allowMultiple"
    @removefile="onRemoveFile"
    @processfileundo="onFileUndo"
    @processfileabort="onFileUndo"
    @init="addFilesToFilepond"
    :server="{
      process: (fieldName, file, metadata, load, error, progress, abort) => {
        uploadImages(file, metadata, load, error, progress, abort)
      },
    }"
  />
.
.
.
onFileUndo(e) {
      console.log('asd')
      console.log(e.detail)
    },
```


```
addFilesToFilepond() {
  const pond = document.querySelector('.filepond--root')
  pond.addEventListener('FilePond:processfileundo', e => {
    console.log('File added', e.detail)
  })
}
```

**Comment by rikschennink** [comment]
> processfileundo doesn't work on vue for me, tried both ways as below. What I would like to have that when the user clicks on undo then it would delete the file (so I'd actually call onRemoveFile for fileundo)

@kepi0809 Please move this to another issue.

**Comment by RicLeP** [comment]
I am using filepond 3.7.4 and when calling processFiles() it processes all files, even those that have already been uploaded.

**Comment by rikschennink** [comment]
@RicLeP Will fix asap, this is probably a side effect of changes to the `processFile` method.

**Comment by rikschennink** [comment]
Just published version `3.7.5` which addresses this issue.

**Comment by leducvin** [comment]
Hi, I am on 4.30.3. Say I load a page with a FilePond instance with already one or more files loaded (from server-side).

What I am seeing is that the `onprocessfiles` callback is not getting called if there are already existing loaded files. `onprocessfile` still gets called when each new file has finished uploading.

If I first delete the loaded files and upload multiple new files, `onprocessfiles` callback gets called.
